Sources/OpenFeature/Provider/MultiProvider/MultiProvider.swift

```diff
@@ -0,0 +1,122 @@
+    }
+
+    public func onContextSet(oldContext: EvaluationContext?, newContext: EvaluationContext) async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for provider in providers {
+                group.addTask {
+                    try await provider.onContextSet(oldContext: oldContext, newContext: newContext)
+                }
+            }
+            try await group.waitForAll()
+        }
+    }
+
+    public func getBooleanEvaluation(key: String, defaultValue: Bool, context: EvaluationContext?) throws
+        -> ProviderEvaluation<Bool>
+    {
+        return try strategy.evaluate(
+            providers: providers,
+            key: key,
+            defaultValue: defaultValue,
+            evaluationContext: context
+        ) { provider in
+            return provider.getBooleanEvaluation(key:defaultValue:context:)
+        }
+    }
+
+    public func getStringEvaluation(key: String, defaultValue: String, context: EvaluationContext?) throws
+        -> ProviderEvaluation<String>
+    {
+        return try strategy.evaluate(
+            providers: providers,
+            key: key,
+            defaultValue: defaultValue,
+            evaluationContext: context
+        ) { provider in
+            return provider.getStringEvaluation(key:defaultValue:context:)
+        }
+    }
+
+    public func getIntegerEvaluation(key: String, defaultValue: Int64, context: EvaluationContext?) throws
+        -> ProviderEvaluation<Int64>
+    {
+        return try strategy.evaluate(
+            providers: providers,
+            key: key,
+            defaultValue: defaultValue,
+            evaluationContext: context
+        ) { provider in
+            return provider.getIntegerEvaluation(key:defaultValue:context:)
+        }
+    }
+
+    public func getDoubleEvaluation(key: String, defaultValue: Double, context: EvaluationContext?) throws
+        -> ProviderEvaluation<Double>
+    {
+        return try strategy.evaluate(
+            providers: providers,
+            key: key,
+            defaultValue: defaultValue,
+            evaluationContext: context
+        ) { provider in
+            return provider.getDoubleEvaluation(key:defaultValue:context:)
+        }
+    }
+
+    public func getObjectEvaluation(key: String, defaultValue: Value, context: EvaluationContext?) throws
+        -> ProviderEvaluation<Value>
+    {
+        return try strategy.evaluate(
+            providers: providers,
+            key: key,
+            defaultValue: defaultValue,
+            evaluationContext: context
+        ) { provider in
+            return provider.getObjectEvaluation(key:defaultValue:context:)
+        }
+    }
+
+    public func observe() -> AnyPublisher<ProviderEvent?, Never> {
+        return Publishers.MergeMany(providers.map { $0.observe() }).eraseToAnyPublisher()
+    }
+
+    public struct MultiProviderMetadata: ProviderMetadata {
+        public var name: String? = MultiProvider.name
```

Comment:
Super nit, but I wonder if this metadata should include the metadata of the internal providers, for better traceability

Reply:
I like that idea. Should I just concatenate the child provider names? like `providers.map { $0.metadata.name }.joined(separator: ", ")`